cytosnake/helpers/datapaths.py

```diff
@@ -0,0 +1,164 @@
+"""
+module: datapath.py
+
+This module contains fuctions that build stirng path that are snakemake compatible.
+"""
+
+import pathlib
+from typing import Optional
+
+import cytosnake.utils.cyto_paths as cyto_paths
+from cytosnake.guards.path_guards import is_valid_path
+from cytosnake.utils.config_utils import load_general_configs, load_meta_path_configs
+
+# constant wildcard name
+WILDCARD_BASENAME = "{basename}"
+
+# loading in config files
+META_PATH_CONFIGS = load_meta_path_configs()
+GENERAL_CONFIGS = load_general_configs()
+
+
+# -----------------------
+# input paths from configs
+# -----------------------
+def get_barcodes() -> str:
+    """Obtains path to the bardcode file within data folder
+
+    Returns
+    -------
+    str
+        return string with barcode name wildcard
+
+    """
+    # Barcodes are optional. If not added, set to "None"
+    try:
+        barcode_path = META_PATH_CONFIGS["project_dir"]["data_directory_contents"][
+            "barcode"
+        ]
+    except KeyError:
+        barcode_path = None
```

Comment:
Depending on the implications, consider a [`defaultdict`](https://docs.python.org/3/library/collections.html#defaultdict-objects) here and elsewhere to assist with defaults where needed.

Reply:
ah, clever! added!